docs/sources/articles/dockerfile_best-practices.md

```diff
@@ -291,28 +291,22 @@ auto-extraction capability, you should always use `COPY`.
 
 ### [`ENTRYPOINT`](https://docs.docker.com/reference/builder/#entrypoint)
 
-The best use for `ENTRYPOINT` is as a helper script. Using `ENTRYPOINT` for
-other tasks can make your code harder to understand. For example,
+The best use for `ENTRYPOINT` is as the main command. 
 
-....docker run -it official-image bash
+Let's start with an example, of an image for the cli tool `s3cmd`:
 
-is much easier to understand than
+    ENTRYPOINT ["s3cmd"]
+    CMD ["--help"]
 
-....docker run -it --entrypoint bash official-image -i
+Now people who consume this image can easily run commands via syntax like the
+following:
 
-This is especially true for new Docker users, who might naturally assume the
```

Comment:
this isn't true. There seems to be a proportion of new users, who like me, read the docs, and the best practices before they start work. Not much different to being told to read the go idioms before you write go code.